app/src/main/java/com/android/shelfLife/viewmodel/recipes/AddRecipeViewModel.kt

```diff
@@ -0,0 +1,277 @@
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import com.android.shelfLife.R
+import com.android.shelfLife.model.foodFacts.FoodUnit
+import com.android.shelfLife.model.foodFacts.Quantity
+import com.android.shelfLife.model.recipe.Ingredient
+import com.android.shelfLife.model.recipe.Recipe
+import com.android.shelfLife.model.recipe.RecipeRepository
+import com.android.shelfLife.model.user.UserRepository
+import com.android.shelfLife.ui.utils.validateNumber
+import com.android.shelfLife.ui.utils.validateString
+import kotlin.time.Duration.Companion.minutes
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+class AddRecipeViewModel(
+    private val recipeRepository: RecipeRepository,
+    private val userRepository: UserRepository
+) : ViewModel() {
+
+  var title by mutableStateOf("")
+  var servings by mutableStateOf("")
+  var time by mutableStateOf("")
+  val ingredients = mutableStateListOf<Ingredient>()
+  var newIngredient by
+      mutableStateOf<Ingredient>(Ingredient(name = "", quantity = Quantity(amount = 0.0)))
+  val instructions = mutableStateListOf<String>()
+  var ingredientName by mutableStateOf("")
+  var ingredientQuantityAmount by mutableStateOf("")
+  var ingredientQuantityUnit by mutableStateOf<FoodUnit>(FoodUnit.GRAM)
+
+  var showIngredientDialog by mutableStateOf(false)
+
+  var error by
+      mutableStateOf<Boolean>(
+          false) // represent a general error (if there is a single error on the screen then this is
+  // true)
+  var errorIngredient by
+      mutableStateOf<Boolean>(false) // represent a general error in the ingredient popUp
+  var titleError by mutableStateOf<Int?>(null)
+  var servingsError by mutableStateOf<Int?>(null)
+  var timeError by mutableStateOf<Int?>(null)
+  var instructionError =
+      mutableStateListOf<
+          Int?>() // a list that the error of each instruction. Their indexes are the same. ex: if
+  // there is an error on the instruction at index 1, the value in this list will be
+  // non-null at index 1
+  var instructionsError by
+      mutableStateOf(
+          false) // a general look at the whole error list, i.e. if there is a single error on the
+  // list, this will be true
+  var ingredientsError by mutableStateOf(false) // a general error for the list of ingredients
+  var ingredientNameError by mutableStateOf<Int?>(null)
+  var ingredientQuantityAmountError by mutableStateOf<Int?>(null)
+
+  init {
+    // we make sure we set the values to their initial values
+    title = ""
+    servings = ""
+    time = ""
+    ingredients.clear()
+    instructions.clear()
+
+    showIngredientDialog = false
+
+    error = false
+    titleError = null
+    servingsError = null
+    timeError = null
+    instructionsError = false
+  }
```

Comment:
This is not necessary, the viewModel gets reinitialized every time the user navigates to the associated composable.